lib/stream.js

```diff
@@ -108,7 +119,8 @@ function createWebSocketStream(ws, options) {
       if (!called) callback(err);
       process.nextTick(emitClose, duplex);
     });
-    ws.terminate();
+
+    if (terminateOnDestroy) ws.terminate();
```

Comment:
I'm not sure how I feel about this. Waiting up to 30s does not seems good, especially because calling `duplex.destroy()` again has no effect. When not using the `Duplex` wrapper users can call `websocket.terminate()` if they do not wait the timeout. They actually can still do that but, meh, they have to use the `WebSocket` object instead of the `Duplex` wrapper.

Reply:
I don't think it's so bad, this looks good to me :+1:

Users shouldn't forcibly close the websocket on error anyway - they should follow proper websocket teardown instead. If they do really want to then it's still possible by using the websocket directly, as you say, and I think it's reasonable that the stream wrapper does correct cleanup by default.

The timeout only comes into play if there's an error and sending the close frame is successful (so there's no underlying socket error) and then the remote peer never replies or closes the socket. Should be rare, impact is limited AFAICT, and closing the socket immediately instead would create the risk of the remote peer losing data we might've already sent (the last paragraph of https://datatracker.ietf.org/doc/html/rfc6455#section-1.4 covers this in more detail).